lib/plugins/aws/deploy/compile/events/apiGateway/lib/methods.js

```diff
@@ -29,6 +29,118 @@ module.exports = {
+              }
+            #end
+
+            {
+              "body": $input.json("$"),
+              "method": "$context.httpMethod",
+              "principalId": "$context.authorizer.principalId",
+              "stage": "$context.stage",
+
+              #set( $map = $input.params().header )
+              "headers": $loop,
+
+              #set( $map = $input.params().querystring )
+              "query": $loop,
+
+              #set( $map = $input.params().path )
+              "path": $loop,
+
+              #set( $map = $context.identity )
+              "identity": $loop,
+
+              #set( $map = $stageVariables )
+              "stageVariables": $loop
+            }
+          `;
+
+          const DEFAULT_FORM_URL_ENCODED_REQUEST_TEMPLATE = `
+            #define( $body )
+              {
+              #foreach( $token in $input.path('$').split('&') )
+                #set( $keyVal = $token.split('=') )
+                #set( $keyValSize = $keyVal.size() )
+                #if( $keyValSize >= 1 )
+                  #set( $key = $util.urlDecode($keyVal[0]) )
+                  #if( $keyValSize >= 2 )
+                    #set( $val = $util.urlDecode($keyVal[1]) )
+                  #else
+                    #set( $val = '' )
+                  #end
+                  "$key": "$val"#if($foreach.hasNext),#end
+                #end
+              #end
+              }
+            #end
+
+            #define( $loop )
+              {
+              #foreach($key in $map.keySet())
+                  "$util.escapeJavaScript($key)":
+                    "$util.escapeJavaScript($map.get($key))"
+                    #if( $foreach.hasNext ) , #end
+              #end
+              }
+            #end
+
+            {
+              "body": $body,
+              "method": "$context.httpMethod",
+              "principalId": "$context.authorizer.principalId",
+              "stage": "$context.stage",
+              
+              #set( $map = $input.params().header )
+              "headers": $loop,
+              
+              #set( $map = $input.params().querystring )
+              "query": $loop,
+              
+              #set( $map = $input.params().path )
+              "path": $loop,
+              
+              #set( $map = $context.identity )
+              "identity": $loop,
+              
+              #set( $map = $stageVariables )
+              "stageVariables": $loop
+            }
+          `;
+
+          const integrationRequestTemplates = {
+            'application/json': DEFAULT_JSON_REQUEST_TEMPLATE,
+            'application/x-www-form-urlencoded': DEFAULT_FORM_URL_ENCODED_REQUEST_TEMPLATE,
+          };
+
+          // check if custom request configuration should be used
+          if (Boolean(event.http.request) === true) {
+            if (typeof event.http.request === 'object') {
+              // merge custom request templates if provided
+              if (event.http.request.template) {
```

Comment:
can you make sure `zipDirectory` is an object as well? otherwise a not-so-friendly error will pop up cause of the iteration.